web/src/main/java/ua/com/vertex/dao/CourseDaoImpl.java

```diff
@@ -26,11 +34,92 @@
     private static final String COLUMN_COURSE_NOTES = "notes";
 
     private final NamedParameterJdbcTemplate jdbcTemplate;
+    private final LogInfo logInfo;
     private static final Logger LOGGER = LogManager.getLogger(CourseDaoImpl.class);
 
-    @Autowired
-    public CourseDaoImpl(DataSource dataSource) {
-        this.jdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
+    @Override
+    public List<Course> getAllCourses(DataNavigator dataNavigator) {
+        List<Course> courses = new ArrayList<>();
+
+        LOGGER.debug("Get all courses list");
+
+        String query = "SELECT c.id, c.name, c.start, c.finished, c.price, c.teacher_name, c.schedule, c.notes " +
+                "FROM Courses c LIMIT :from, :offset";
+
+        MapSqlParameterSource parameters = new MapSqlParameterSource();
+        parameters.addValue("from", (dataNavigator.getCurrentNumberPage() - 1) * dataNavigator.getCurrentRowPerPage());
+        parameters.addValue("offset", dataNavigator.getCurrentRowPerPage());
+
+        try {
+            courses = jdbcTemplate.query(query, parameters, new CourseDaoImpl.CoursesRowMapping());
+        } catch (EmptyResultDataAccessException e) {
+            LOGGER.warn("Something went wrong", e);
+        }
+        return courses;
+    }
+
+    private static final class CoursesRowMapping implements RowMapper<Course> {
+        public Course mapRow(ResultSet resultSet, int i) throws SQLException {
+            Course course = new Course();
+            course.setId(resultSet.getInt("id"));
+            course.setName(resultSet.getString("name"));
+            course.setStart(resultSet.getDate("start").toLocalDate());
+            course.setFinished(resultSet.getInt("finished") != 0);
+            course.setPrice(resultSet.getBigDecimal("price"));
+            course.setTeacherName(resultSet.getString("teacher_name"));
+            course.setSchedule(resultSet.getString("schedule"));
+            course.setNotes(resultSet.getString("notes"));
+            return course;
+        }
+    }
+
+    @Override
+    public int getQuantityCourses() throws SQLException {
+        LOGGER.debug("Get all courses list");
```

Comment:
you write that you are going to get a list, but getting a quantity. It makes me sad =(